### PR TITLE
Improvements/ux ui

### PR DESCRIPTION
**Pull request by doliG**
Changes:

- [Ux/Ui] Separate delete button from others,
- [Ux] Most clicked button moved on the left,
- [Ux/Ui] MailCard become MailList, on line, a more compact and readable alternative,
- [Ux] Better formatting for date on MailList

image

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "axios": "^0.19.0",
     "bootstrap": "^4.3.1",
     "copy-to-clipboard": "^3.2.0",
+    "dayjs": "^1.8.16",
```
Comment (lodi-g) [inline web/package.json:9]: It looks like you've included dayjs, probably to replace moment & have a smaller bundle size.
It's a good idea, however moment is still a dependency and still used everywhere else.

Could you please either swap moment for dayjs in this / another PR or remove the dep?

Thank you!

**Review by lodi-g — CHANGES_REQUESTED** [review]

File: web/package.json (hunk 1 above)
Comment (doliG) [reply web/package.json:9]: I tried to replace moment. Sadly DayJS haven’t the very useful “fromNow” method. 
Looks like I forgot to uninstall dayjs, nice shot. Gonna remove it in a next commit .

Comment (lodi-g) [reply web/package.json:9]: It seems to exists under the name 'relativeTime': https://github.com/iamkun/dayjs/blob/master/docs/en/Plugin.md#relativetime

Comment (doliG) [reply web/package.json:9]: Done. Please check new commit.

File: db/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (lodi-g) [inline db/package-lock.json]: What is this file/why is this here?

File: web/src/components/MailBox/MailItem.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import React from 'react'
+import dayjs from 'dayjs'
+import relativeTime from 'dayjs/plugin/relativeTime'
+import Form from 'react-bootstrap/Form'
+import ListGroup from 'react-bootstrap/ListGroup'
+import Row from 'react-bootstrap/Row'
+import Col from 'react-bootstrap/Col'
+
+const MailItem = ({ id, subject, receivedOn, from, read, selected, onSelect, onClick }) => {
+  dayjs.extend(relativeTime)
```
Comment (lodi-g) [inline web/src/components/MailBox/MailItem.jsx]: Wouldn't it be simpler to move the initialization above the component so it doesn't re initialize each time the component is rendered?

File: web/src/routes/Mail.jsx [hunk 4]
```diff
@@ -27,6 +28,10 @@ class Mail extends React.Component {
     this.setState({ redirect: true })
   }
 
+  componentWillMount() {
+    dayjs.extend(relativeTime)
```
Comment (lodi-g) [inline web/src/routes/Mail.jsx]: Same as MailItem:10

**Review by lodi-g — CHANGES_REQUESTED** [review]

**Comment by lodi-g** [comment]
Actually you could move the `dayjs.extend(relativeTime)` to the index.js (entry point) of the app I guess?

File: db/package-lock.json (hunk 2 above)
Comment (doliG) [reply db/package-lock.json]: I've removed it and force pushed

File: web/src/components/MailBox/MailItem.jsx (hunk 3 above)
Comment (doliG) [reply web/src/components/MailBox/MailItem.jsx]: You're right. It's done. I let you resolve and merge.

File: web/package.json [hunk 5]
```diff
@@ -6,7 +6,6 @@
     "axios": "^0.19.0",
     "bootstrap": "^4.3.1",
     "copy-to-clipboard": "^3.2.0",
-    "dayjs": "^1.8.16",
```
Comment (lodi-g) [inline web/package.json]: You should let dayjs and remove moment, not the other way :stuck_out_tongue:

**Review by lodi-g — CHANGES_REQUESTED** [review]

File: web/package.json (hunk 5 above)
Comment (doliG) [reply web/package.json]: Outdated change

Comment (doliG) [reply web/package.json]: Aw got it, I rebase and fucked up this one. Well, nice catch lol

**Comment by doliG** [comment]
Is it good now ?
